components/ActionSheet/tests/snapshot/MDCActionSheetControllerSnapshotTests.m

```diff
@@ -968,4 +984,105 @@ - (void)testActionSheetWithHeaderShownButNoTitleOrMessage {
+    self.actionSheetController.actionFont = bodyFont;
+    self.actionSheetController.view.bounds = CGRectMake(0, 0, 1000, 1000);
+
+    // Create a window so the Action Sheet's view can inherit the trait environment.
+    MDCActionSheetControllerCustomTraitCollectionTestsWindowFake *window =
+        [[MDCActionSheetControllerCustomTraitCollectionTestsWindowFake alloc] init];
+    [window addSubview:self.actionSheetController.view];
+
+    // When
+    window.traitCollectionOverride =
+        [UITraitCollection traitCollectionWithPreferredContentSizeCategory:
+                               UIContentSizeCategoryAccessibilityExtraExtraExtraLarge];
+    [window traitCollectionDidChange:nil];
+    [self.actionSheetController.view layoutIfNeeded];
+
+    // Then
+    [self snapshotVerifyView:self.actionSheetController.view];
+  }
+}
+
+/**
+ Test that @c adjustsFontForContentSizeCategory will scale an appropriate font to a smaller size
+ when the preferred content size category decreases.
+ */
+- (void)testAdjustsFontForContentSizeDownscalesUIFontMetricsFontsForSizeCategoryXS {
+  if (@available(iOS 11.0, *)) {
+    // Given
+    self.actionSheetController = [MDCActionSheetController actionSheetControllerWithTitle:@"Foo"
+                                                                                  message:@"Bar"];
+    self.actionSheetController.adjustsFontForContentSizeCategory = YES;
+    MDCActionSheetAction *action = [MDCActionSheetAction actionWithTitle:kShortTitle1Latin
+                                                                   image:nil
+                                                                 handler:nil];
+    [self.actionSheetController addAction:action];
+    UIFont *originalFont = [UIFont fontWithName:@"Zapfino" size:20];
+    UITraitCollection *extraSmallTraits = [UITraitCollection
+        traitCollectionWithPreferredContentSizeCategory:UIContentSizeCategoryExtraSmall];
+    // Setup title font
+    UIFontMetrics *titleMetrics = [UIFontMetrics metricsForTextStyle:UIFontTextStyleTitle1];
+    UIFont *titleFont = [titleMetrics scaledFontForFont:originalFont
+                          compatibleWithTraitCollection:extraSmallTraits];
+    self.actionSheetController.titleFont = titleFont;
+    // Setup message font
+    UIFontMetrics *messageMetrics = [UIFontMetrics metricsForTextStyle:UIFontTextStyleSubheadline];
+    UIFont *messageFont = [messageMetrics scaledFontForFont:originalFont
+                              compatibleWithTraitCollection:extraSmallTraits];
+    self.actionSheetController.messageFont = messageFont;
+    // Setup actions font
+    UIFontMetrics *bodyMetrics = [UIFontMetrics metricsForTextStyle:UIFontTextStyleBody];
+    UIFont *bodyFont = [bodyMetrics scaledFontForFont:originalFont
+                        compatibleWithTraitCollection:extraSmallTraits];
+    self.actionSheetController.actionFont = bodyFont;
+    self.actionSheetController.view.bounds = CGRectMake(0, 0, 1000, 1000);
```

Comment:
Tip: Is there any reason why these snapshots need to be 1000x1000? Can you not use the preferredContentSize or something similar to size the view?